test/integration/identity/stateTransition/assetLockProof/chain/validateChainAssetLockProofStructureFactory.spec.js

```diff
@@ -0,0 +1,342 @@
+
+      expectJsonSchemaError(result, 2);
+
+      const [error, byteArrayError] = result.getErrors();
+
+      expect(error.dataPath).to.equal('.outPoint[0]');
+      expect(error.keyword).to.equal('type');
+
+      expect(byteArrayError.keyword).to.equal('byteArray');
+
+      expect(stateRepositoryMock.fetchTransaction).to.not.be.called();
+      expect(validateAssetLockTransactionMock).to.not.be.called();
+      expect(stateRepositoryMock.fetchLatestPlatformBlockHeader).to.not.be.called();
+    });
+
+    it('should not be shorter than 36 bytes', async () => {
+      rawProof.outPoint = Buffer.alloc(35);
+
+      const result = await validateChainAssetLockProofStructure(rawProof);
+
+      expectJsonSchemaError(result);
+
+      const [error] = result.getErrors();
+
+      expect(error.dataPath).to.equal('.outPoint');
+      expect(error.keyword).to.equal('minItems');
+
+      expect(stateRepositoryMock.fetchTransaction).to.not.be.called();
+      expect(validateAssetLockTransactionMock).to.not.be.called();
+      expect(stateRepositoryMock.fetchLatestPlatformBlockHeader).to.not.be.called();
+    });
+
+    it('should not be longer than 36 bytes', async () => {
+      rawProof.outPoint = Buffer.alloc(37);
+
+      const result = await validateChainAssetLockProofStructure(rawProof);
+
+      expectJsonSchemaError(result);
+
+      const [error] = result.getErrors();
+
+      expect(error.dataPath).to.equal('.outPoint');
+      expect(error.keyword).to.equal('maxItems');
+
+      expect(stateRepositoryMock.fetchTransaction).to.not.be.called();
+      expect(validateAssetLockTransactionMock).to.not.be.called();
+      expect(stateRepositoryMock.fetchLatestPlatformBlockHeader).to.not.be.called();
+    });
+
+    it('should point to existing transaction', async () => {
+      stateRepositoryMock.fetchTransaction.resolves(null);
+
+      const result = await validateChainAssetLockProofStructure(rawProof);
+
+      expectValidationError(result, IdentityAssetLockTransactionIsNotFoundError);
+      const [error] = result.getErrors();
+
+      expect(error.getOutPoint()).to.deep.equal(rawProof.outPoint);
+
+      expect(stateRepositoryMock.fetchTransaction).to.be.calledOnceWithExactly(
+        '6e200d059fb567ba19e92f5c2dcd3dde522fd4e0a50af223752db16158dabb1d',
+      );
+      expect(validateAssetLockTransactionMock).to.not.be.called();
+      expect(stateRepositoryMock.fetchLatestPlatformBlockHeader).to.be.calledOnce();
+    });
+
+    it('should point to valid transaction', async () => {
+      const consensusError = new ConsensusError('test');
+      validateAssetLockTransactionResult.addError(consensusError);
+
+      const result = await validateChainAssetLockProofStructure(rawProof);
+
+      expectValidationError(result, ConsensusError);
+      const [error] = result.getErrors();
+
+      expect(error).to.deep.equal(consensusError);
+    });
+
+    it('should has valid height', async () => {
```

Comment:
`should point to transaction from block lower than core chain locked height`

tests must be different